src/glb_main.c

```diff
@@ -30,7 +34,13 @@ allocate_resources(const glb_cnf_t* conf,
                    int* ctrl_sock,
                    int* listen_sock)
 {
-    if (mkfifo (conf->fifo_name, S_IRUSR | S_IWUSR)) {
+    gid_t gid = getegid();
+    struct group *grp = getgrnam(RUN_AS_USER);  
+    setegid(grp->gr_gid);
+    mode_t um = umask(0); 
+    if (mkfifo (conf->fifo_name, S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP)) {
+        setegid(gid);
+        umask(um);
```

Comment:
why bother doing `setegid()`/`umask()` in this case? If it is needed regardless of `mkfifo()` outcome, then it should be done outside of `if()`